### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by biscottigelato** [btcsuite/btcd#1219]
I am getting index out of range simply by running btcd. 

```
howardw_lee@instance-1:~$ panic: runtime error: index out of range
goroutine 189 [running]:
main.(*serverPeer).OnGetCFCheckpt(0xc42018c5a0, 0xc434cfe600, 0xc432038270)
        /home/howardw_lee/gocode/src/github.com/btcsuite/btcd/server.go:943 +0xce2
main.(*serverPeer).OnGetCFCheckpt-fm(0xc434cfe600, 0xc432038270)
        /home/howardw_lee/gocode/src/github.com/btcsuite/btcd/server.go:1846 +0x3e
github.com/btcsuite/btcd/peer.(*Peer).inHandler(0xc434cfe600)
        /home/howardw_lee/gocode/src/github.com/btcsuite/btcd/peer/peer.go:1617 +0x1013
created by github.com/btcsuite/btcd/peer.(*Peer).start
        /home/howardw_lee/gocode/src/github.com/btcsuite/btcd/peer/peer.go:2079 +0x2da
```

This is at origin/master - fdfc19097e7ac6b57035062056f5b7b4638b8898

btcd.conf is as attached & btcd.log are as attached
[btcd.conf.txt](https://github.com/btcsuite/btcd/files/2195410/btcd.conf.txt)
[btcd.log](https://github.com/btcsuite/btcd/files/2195411/btcd.log)

**Comment by JimmyMow** [comment]
I am getting this as well

**Comment by biscottigelato** [comment]
I think this has something to do with some newer version of LND requesting an unsupported neutrino filter type from the BTCD node

**Comment by Roasbeef** [comment]
Fixed with 5e86c374110b3974675669c57e046deed0d70a85, also have a follow up patch coming to improve the concurrency.
